mitol-django-oauth-toolkit-extensions/tests/test_backends.py

```diff
@@ -0,0 +1,48 @@
+"""Backends tests"""
+import pytest
+
+from mitol.oauth_toolkit_extensions.backends import ApplicationAccessOrSettingsScopes
+
+
+@pytest.mark.parametrize(
+    "application_exists, is_configured", [(True, False), (True, True), (False, False),]
+)
+def test_application_access_settings_scopes_get_available_scopes(
+    mocker, application_exists, is_configured
+):
+    """Verify get_available_scopes() uses ApplicationAccess or defaults to settings"""
+    application = mocker.Mock() if application_exists else None
+    if application is not None:
+        application.access = (
+            mocker.Mock(scopes_list=["limited"]) if is_configured else None
+        )
+
+    scopes = ApplicationAccessOrSettingsScopes()
+    if is_configured:
+        assert scopes.get_available_scopes(application=application) == ["limited"]
+    else:
+        assert scopes.get_available_scopes(application=application) == [
+            "read",
+            "write",
+            "only_logout",
+        ]
```

Comment:
Some ideas for improving these test cases:
1. Combine them into 1. The two methods we're testing have the same signature and I think we could expect that to remain true
2. Use the OAuth settings values to make the expectation more explicit

```python
@pytest.mark.parametrize(
    "application_exists, is_configured", [(True, False), (True, True), (False, False), ]
)
def test_application_access_settings_scopes(
        mocker, settings, application_exists, is_configured
):
    """Verify get_available_scopes() uses ApplicationAccess or defaults to settings"""
    application_access_scopes = ["limited"]
    # Might need to sort this...
    settings_available_scopes = settings.OAUTH2_PROVIDER['SCOPES'].keys()
    settings_default_scopes = settings.OAUTH2_PROVIDER['DEFAULT_SCOPES']
    if application_exists:
        application = mocker.Mock(
            access=(
                mocker.Mock(scopes_list=application_access_scopes) if is_configured else None
            )
        )
    else:
        application = None

    scopes = ApplicationAccessOrSettingsScopes()
    if is_configured:
        assert scopes.get_available_scopes(application=application) == application_access_scopes
        assert scopes.get_default_scopes(application=application) == application_access_scopes
    else:
        assert scopes.get_available_scopes(application=application) == settings_available_scopes
        assert scopes.get_default_scopes(application=application) == settings_default_scopes
```